### PR TITLE
HPCC4J-574 Jirabot improvements

### PR DESCRIPTION
**Pull request by jpmcmu**
- Moved github user dictionary to github action var
- Changed logic for changing jira issue user

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [X] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

File: .github/workflows/Jirabot.yml [hunk 1]
```diff
@@ -37,10 +37,11 @@ jobs:
           PULL_URL: ${{ github.event.pull_request.html_url }}
           COMMENTS_URL: ${{ github.event.pull_request.comments_url }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
```
Comment (rpastrana) [inline .github/workflows/Jirabot.yml:40]: GITHUB_JIRA_USER_MAP ?

**Review by rpastrana — COMMENTED** [review]
@jpmcmu a couple of var refactoring as mentioned offline.

**Comment by jpmcmu** [comment]
@rpastrana Implemented code review changes and rebased to 8.12.x

File: .github/workflows/Jirabot.yml [hunk 2]
```diff
@@ -24,79 +24,113 @@ jobs:
...
+                elif statusName == 'In Progress':
+                    transition = ''
+                elif statusName == 'Resolved':
+                    transition = 'Reopen Issue'
+                elif statusName == 'Closed':
+                    transition = 'Reopen Issue'
+                else:
+                    transition = ''
+
+                if transition != '':
```
Comment (rpastrana) [inline .github/workflows/Jirabot.yml:62]: what happens if transition == '' ?

**Review by rpastrana — APPROVED** [review]

File: .github/workflows/Jirabot.yml [hunk 3]
```diff
@@ -24,79 +24,113 @@ jobs:
...
+                if statusName == 'Open':
+                    transition = 'Start Progress'
+                elif statusName == 'In Progress':
+                    transition = ''
+                elif statusName == 'Resolved':
+                    transition = 'Reopen Issue'
+                elif statusName == 'Closed':
+                    transition = 'Reopen Issue'
+                else:
+                    transition = ''
```
Comment (rpastrana) [inline .github/workflows/Jirabot.yml:60]: Let's log this else condition, something like "Unexpected Jira status encountered: statusName"

File: .github/workflows/Jirabot.yml [hunk 4]
```diff
@@ -67,10 +68,17 @@ jobs:
                         transitions = jira.transitions(issue)
                         result += 'Error: Transition: "' + transition + '" failed with: "' + str(error) + '" Valid transitions=' + str(transitions) + '\n'
 
-                if issue.fields.customfield_10010 is None:
-                    issue.update(fields={'customfield_10010': pull_url})
+                prFieldName = propertyMap.get('pullRequestFieldName', 'customfield_10010')
```
Comment (rpastrana) [inline .github/workflows/Jirabot.yml:71]: let's target the new jira name as a long term solution

**Review by rpastrana — APPROVED** [review]
@jpmcmu approved, we'll change the default pr custom field name once migrated.

**Comment by jpmcmu** [comment]
@rpastrana Finished testing the issue property map change, looks good. Squashed and ready to merge if it still looks good

**Pull request merged**